module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -304,15 +326,38 @@ protected function renewTenantToken()
      */
     protected function checkTenantToken()
     {
-        $response = $this->makeRequest('GET', '/users', [], [], [401, 403]);
-        if ($response->getStatusCode() >= 400) {
+        if ($this->useLegacyAuthentication()) {
+            $response = $this->makeRequest('GET', '/users', [], [], [401, 403]);
+            if ($response->getStatusCode() >= 400) {
+                $this->token = null;
+                $this->tokenExpiration = null;
+                $this->renewTenantToken();
+                return false;
+            }
+            return true;
+        }
+        if ($this->checkTenantTokenExpired()) {
             $this->token = null;
+            $this->tokenExpiration = null;
             $this->renewTenantToken();
             return false;
         }
         return true;
     }
 
+    /**
+     * Check if our token has expired. Return true if it has expired, false if it has not.
+     *
+     * @return bool
+     */
+    protected function checkTenantTokenExpired()
+    {
+        return
+            $this->token == null ||
+            $this->tokenExpiration == null ||
+            strtotime('now') >= strtotime($this->tokenExpiration);
```

Comment:
Another minor style thing -- we generally prefer leading rather than trailing operators; it helps clarify how lines relate to the ones above them.

```suggestion
            $this->token == null
            || $this->tokenExpiration == null
            || strtotime('now') >= strtotime($this->tokenExpiration);
```

Reply:
Good to know, I'll try to remember that for future PRs. I've made that update.